File: problems/Codeforces/58A/solution.cpp
#include <iostream>
#include <iterator>
#include <vector>
#include <print>

int main() {
    using namespace std;
    vector<char> s{istreambuf_iterator<char>{cin}, {}};
    vector<char> hello = {'h','e','l','l','o'};
    int i = 0;
    for(char c: s) {
        if(i == 5) break;
        if(c == hello[i]) i++;
    }
    print("{}", i == 5 ? "YES" : "NO");
    return 0;
}

File: problems/Codeforces/58A/solution.rs
use std::io::{self, Read};

fn main() -> io::Result<()> {
    let mut input = Vec::new();
    io::stdin().lock().read_to_end(&mut input)?;
    let hello = [b'h',b'e',b'l',b'l',b'o'];
    let mut i = 0;
    for &c in &input {
        if i == 5 {
            break;
        }
        if c == hello[i] {
            i += 1;
        }
    }
    print!("{}", if i == 5 {"YES"} else {"NO"});
    Ok(())
}
